client/my-sites/domains/paths.js

```diff
@@ -61,8 +61,19 @@ export function domainAddNew( siteName, searchTerm ) {
 	return path;
 }
 
-export function domainAddEmailUpsell( siteName, domainName ) {
-	return `/domains/add/${ domainName }/email/${ siteName }`;
+/**
+ * Builds a URL query string from an object. Handles null values.
+ *
+ * @param {Object} parameters - optional path prefix
+ * @returns {string} the corresponding query string
```

Comment:
The comment for the `parameters` parameter doesn't sound right. I think we could also refine this whole docblock comment a bit, e.g.:

```suggestion
 * Builds a URL query string with the specified parameters.
 *
 * @param {Object} parameters - set of parameters, any null one will be skipped
 * @returns {string} the corresponding query string prepended with a question mark
```